lib/fs.js

```diff
@@ -62,13 +74,8 @@ const DEBUG = process.env.NODE_DEBUG && /fs/.test(process.env.NODE_DEBUG);
 const errnoException = util._errnoException;
 
 function getOptions(options, defaultOptions) {
-  if (options === null || options === undefined ||
-      typeof options === 'function') {
-    return defaultOptions;
-  }
-
   if (typeof options === 'string') {
-    defaultOptions = util._extend({}, defaultOptions);
+    defaultOptions = shallowClone(defaultOptions);
```

Comment:
Do we have any benchmarks comparing `util._extend` `Object.assign` and `shallowClone`?

Would it be faster to do `defaultOptions = Object.create(defaultOptions)` and use the fact it's on the prototype?